src/Aspire.Hosting/Dashboard/KubernetesDataSource.cs

```diff
@@ -0,0 +1,581 @@
+
+                case ResourceKind.Executable:
+                    if (_executablesMap.TryGetValue(resourceName, out var executable))
+                    {
+                        if (executable.IsCSharpProject())
+                        {
+                            // Project
+                            var projectViewModel = ConvertToProjectViewModel(executable);
+
+                            await _onResourceChanged(projectViewModel, ObjectChangeType.Upsert).ConfigureAwait(false);
+                        }
+                        else
+                        {
+                            // Executable
+                            var executableViewModel = ConvertToExecutableViewModel(executable);
+
+                            await _onResourceChanged(executableViewModel, ObjectChangeType.Upsert).ConfigureAwait(false);
+                        }
+                    }
+                    break;
+            }
+        }
+    }
+
+    private ContainerViewModel ConvertToContainerViewModel(Container container)
+    {
+        var model = new ContainerViewModel
+        {
+            Name = container.Metadata.Name,
+            DisplayName = container.Metadata.Name,
+            Uid = container.Metadata.Uid,
+            ContainerId = container.Status?.ContainerId,
+            CreationTimeStamp = container.Metadata.CreationTimestamp?.ToLocalTime(),
+            Image = container.Spec.Image!,
+            LogSource = new DockerContainerLogSource(container.Status!.ContainerId!),
+            State = container.Status?.State,
+            ExpectedEndpointsCount = GetExpectedEndpointsCount(_servicesMap.Values, container),
+            Command = container.Spec.Command,
+            Args = container.Spec.Args
+        };
+
+        if (container.Spec.Ports != null)
+        {
+            foreach (var port in container.Spec.Ports)
+            {
+                if (port.ContainerPort != null)
+                {
+                    model.Ports.Add(port.ContainerPort.Value);
+                }
+            }
+        }
+
+        FillEndpoints(container, model, ResourceKind.Container);
+
+        if (model.ContainerId is not null && _containerIdsHavingDockerInspections.Add(model.ContainerId))
+        {
+            // This container has not yet been inspected. Call kubernetes on the CLI to obtain the environment
+            // for this container. When returned, the values will be cached in _dockerEnvironmentByContainerId
+            // and an updated container resource published, which will pick up the docker environment.
+            Task.Run(() => ComputeEnvironmentVariablesFromDocker(model.ContainerId, container.Metadata.Name));
+        }
+
+        if (model.ContainerId is not null && _dockerEnvironmentByContainerId.TryGetValue(model.ContainerId, out var dockerEnvironment))
+        {
+            FillEnvironmentVariables(model.Environment, dockerEnvironment, null);
+        }
+        else if (container.Spec.Env is not null)
```

Comment:
Containers now have `EffectiveEnv` too https://github.com/dotnet/aspire/issues/1114

Reply:
What's the logic here? Should we replace use of `Env` with `EffectiveEnv`, or some other approach?

Reply:
Yes, just replacing the use of `Env` with `EffectiveEnv` should do the trick, with the caveat that `Env` is part of the spec, so it is controlled by the AppHost and never changes, vs. `EffectiveEnv` is computed at startup and initially empty.

Reply:
I don't see `ContainerSpec.EffectiveEnv` in the code. Perhaps it needs to be mapped in the JSON types. Is it on the spec or the status object? For executables it's on the status.

Perhaps this also means we would no longer need `ComputeEnvironmentVariablesFromDocker` and the process that spins off processes to query the env via the CLI.

Reply:
I think I worked this out. Thanks.